src/sgame/sg_bot_util.cpp

```diff
@@ -1844,39 +1858,15 @@ void BotBuyUpgrade( gentity_t *self, upgrade_t upgrade )
 
 		if ( upgrade == UP_LIGHTARMOUR )
 		{
-			if ( !G_RoomForClassChange( self, PCL_HUMAN_LIGHT, newOrigin ) )
-			{
-				return;
-			}
-			VectorCopy( newOrigin, self->client->ps.origin );
-			self->client->ps.stats[ STAT_CLASS ] = PCL_HUMAN_LIGHT;
-			self->client->pers.classSelection = PCL_HUMAN_LIGHT;
-			BotSetNavmesh( self, PCL_HUMAN_LIGHT );
-			self->client->ps.eFlags ^= EF_TELEPORT_BIT;
+			BotChangeClass( self, PCL_HUMAN_LIGHT, newOrigin );
```

Comment:
The new code is not equivalent. It does not return from `BotBuyUpgrade` when there is no room.